tensorflow_serving/model_servers/prediction_service_impl.cc

```diff
@@ -51,18 +52,24 @@ thread::ThreadPoolOptions GetThreadPoolOptions(
 ::grpc::Status PredictionServiceImpl::Predict(::grpc::ServerContext *context,
                                               const PredictRequest *request,
                                               PredictResponse *response) {
+  const uint64 start = Env::Default()->NowMicros();
   tensorflow::RunOptions run_options = tensorflow::RunOptions();
   if (enforce_session_run_timeout_) {
     run_options.set_timeout_in_ms(
         DeadlineToTimeoutMillis(context->raw_deadline()));
   }
 
-  const ::grpc::Status status =
-      ToGRPCStatus(predictor_->Predict(run_options, core_, *request, response));
+  const ::tensorflow::Status tf_status = predictor_->Predict(
+      run_options, core_, *request, response);
+  const ::grpc::Status status = ToGRPCStatus(tf_status);
 
   if (!status.ok()) {
     VLOG(1) << "Predict failed: " << status.error_message();
   }
+  RecordRequestLatency(request->model_spec().name(), /*api=*/"Predict",
```

Comment:
(here and elsewhere), we do not want to record latency for requests that error out. no?

Reply:
OK, I try to add status label in request_latency.